Made hidden ProjectSettings groups more explicit

KoBeWi opened the pull request:
Follow-up to #61751
Changes long `if` into a more clean list of prefixes.

I experimented with autoloads. At first I tried marking them as internal when added, but the problem is that this isn't saved. The properties are simply loaded with ProjectSettings resource and then re-flagged. So there is no avoiding `_get_property_list()` or `_validate_property()` without some bigger changes that just aren't worth it.

KoBeWi committed:
Made hidden ProjectSettings groups more explicit

akien-mga commented on core/config/project_settings.cpp:
```diff
@@ -1242,10 +1257,16 @@ ProjectSettings::ProjectSettings() {
 	GLOBAL_DEF_INTERNAL("internationalization/locale/translation_remaps", PackedStringArray());
 	GLOBAL_DEF_INTERNAL("internationalization/locale/translations", PackedStringArray());
+
+	add_hidden_prefix("input/");
+	add_hidden_prefix("importer_defaults/");
+	add_hidden_prefix("autoload/");
+	add_hidden_prefix("editor_plugins/");
+	add_hidden_prefix("shader_globals/");
```
My idea was that `add_hidden_prefix` could be called from the actual code which adds stuff to these categories (assuming that for each of those it's done in a single class). But it's fine as is too I guess, and it's still available for plugin use.

KoBeWi replied:
I can move these to their relevant code parts, but it will make them more difficult to find. But I guess it's not that important 🤔

KoBeWi commented:
Rebased. I moved the prefix definition to their respective part of code. I also changed the prefix container from List to LocalVector, which I think is faster.

Calinou reviewed (approved):
Code looks good to me.

Note that this doesn't change the editor UI in any way, even when Advanced Settings is enabled.

AThousandShips commented:
It appears to add a lot of extra data to the docs, something broken in the implementation?

KoBeWi commented:
Fixed, I had to move `input/` prefix back to ProjectSettings.

This is a funny problem. InputMap is created before ProjectSettings, so `add_hidden_prefix()` can't be used in its constructor. The 2 earliest methods InputMap are `load_default()`, which is called once in Main when the editor starts, and `load_from_project_settings()`, which is also called in Main, but is also a bound method, so adding prefix there would cause problems. Apparently doctool runs in non-editor context, hence calling the prefix method in `load_default()` did not work correctly. I think the best fix would be creating InputMap after ProjectSettings, but it's risky, so I just made an exception.

akien-mga reviewed (approved):
Looks much cleaner.

akien-mga commented:
Thanks!
